merino-suggest/src/lib.rs

```diff
@@ -291,6 +291,31 @@ pub trait SuggestionProvider: Send + Sync {
     fn is_null(&self) -> bool {
         false
     }
+
+    /// Generate a set of cache inputs for a given query specific to this
+    /// provider. Any property of the query that affects how suggestions are
+    /// generated should be included.
+    ///
+    /// By default, all properties of the query are used, but providers should
+    /// narrow this to a smaller scope.
+    fn cache_inputs(&self, req: &SuggestionRequest, cache_inputs: &mut dyn CacheInputs) {
+        cache_inputs.add(req.query.as_bytes());
+        cache_inputs.add(&[req.accepts_english as u8]);
+        cache_inputs.add(format!("{:?}", req.country).as_bytes());
```

Comment:
We could replace those `format!`s with `cache_inputs.add(req.country.as_deref().unwrap_or("None").as_bytes())`.

I know it's messy, but my "let's-save-one-allocation" OCD just kicked in. 😂